### DISCUSSION TITLE
How to improve efficiency in data preprocessing?

### DISCUSSION THREAD

**Discussion by FFFFFG-IGG** [groops-devs/groops#107] [Q&A] [answered]
Hello, when I was doing Network analysis with about 293 IGS stations, I found that preprocessing observation files to groops-format very time-consuming, which took about 8 hours. What's worse, when setting sampling of 30s and disabling estimation of satellite clock , orbit ,attitude and signal bias as well as VTEC parameters, the network analysis took hours to finish, which may lower the time-efficiency for getting position or Troposphere values. Could you give me some suggestions for improving efficiency?

**Comment by tmayerguerr** [comment, marked as answer]
GROOPS programs can be run in parallel on multiple processor cores ([Parallelization](https://groops-devs.github.io/groops/html/general.parallelization.html)). Especially [LoopProgram](https://groops-devs.github.io/groops/html/LoopPrograms.html) can be parallized if processCountPerIteration=1. Under Windows GROOPS can be started with 4 processes with the following command
`mpiexec -n 4 groopsMPI.exe`
regards
Torsten

**Reply by FFFFFG-IGG** [reply]
Thanks for your quick reply.And I have try parallel computation in groops in the case of NetAnalysis(7 days, 197 IGS stations, 5-minutes sampling) using 'mpiexec -n 4 groopsMPI.exe' , same as you mentioned before. But I found that parallel mode(4 cores, take about 9 hours) is more time-consuming than non-parallel mode(one core, take about 5 hours). It seems weird, could you give me some illustration about that?
regards,
Yongzhao Fan

**Reply by FFFFFG-IGG** [reply]
I've gotten that solution in discussion #128
